### PR TITLE
Add desired phase plan lib

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
Add desired phase plan library to serialize and reserialize the message.

## Related Issue
NA

## Motivation and Context
UC3

## How Has This Been Tested?
Unit test

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .sonarqube/sonar-scanner.properties [hunk 1]
```diff
@@ -30,7 +30,8 @@ sonar.coverageReportPaths= /home/carma-streets/kafka_clients/coverage/coverage.x
 /home/carma-streets/streets_utils/streets_vehicle_list/coverage/coverage.xml, \
 /home/carma-streets/streets_utils/streets_signal_phase_and_timing/coverage/coverage.xml, \
 /home/carma-streets/tsc_client_service/coverage/coverage.xml, \
-/home/carma-streets/streets_utils/streets_vehicle_scheduler/coverage/coverage.xml
+/home/carma-streets/streets_utils/streets_vehicle_scheduler/coverage/coverage.xml, \
+/home/carma-streets/streets_utils/streets_desired_phase_plan/coverage/coverage.xml
 #/home/carma-streets/message_services/coverage/coverage.xml, 
 #Encoding of the source code. Default is default system encoding
```
Comment (paulbourelly999) [inline .sonarqube/sonar-scanner.properties:36]: You need to add the src and test files for your new library.

Comment (dan-du-car) [reply .sonarqube/sonar-scanner.properties:36]: Is this about coverage? The test are scanned in the coverage.sh script.

Comment (dan-du-car) [reply .sonarqube/sonar-scanner.properties:36]: updated build.shhttps://github.com/usdot-fhwa-stol/carma-streets/pull/188/commits/0d175a50781728249cfc8d256d440adfafae0fc6

File: streets_utils/streets_desired_phase_plan/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Streets Desired phase plan
+
+## Introduction
+
+This CARMA-Streets library is meant to handle JSON serialization and deserialization for the CARMA-Streets internal Desired Phase Plan JSON message.Below is a sample Desired Phase Plan JSON message created by this library.
```
Comment (paulbourelly999) [inline streets_utils/streets_desired_phase_plan/README.md:5]: We probably can include more information about what this message is for. Who creates it, who consumes it and what is done with that information. Maybe even information about when it is produced.

File: streets_utils/streets_desired_phase_plan/cmake/streets_desired_phase_plan_libConfig.cmake.in [hunk 3]
```diff
@@ -0,0 +1,10 @@
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+find_dependency(spdlog REQUIRED)
+find_dependency(RapidJSON REQUIRED)
+find_dependency(GTest REQUIRED)
```
Comment (paulbourelly999) [inline streets_utils/streets_desired_phase_plan/cmake/streets_desired_phase_plan_libConfig.cmake.in:6]: I don't think you need to include GTest as a dependency unless your library depends on GTest

**Review by paulbourelly999 — COMMENTED** [review]
Looks good mostly. If we make small improvements to the documentation I am good to merge this.

File: streets_utils/streets_desired_phase_plan/cmake/streets_desired_phase_plan_libConfig.cmake.in (hunk 3 above)
Comment (dan-du-car) [reply streets_utils/streets_desired_phase_plan/cmake/streets_desired_phase_plan_libConfig.cmake.in:6]: There are unit tests

File: streets_utils/streets_desired_phase_plan/README.md (hunk 2 above)
Comment (dan-du-car) [reply streets_utils/streets_desired_phase_plan/README.md:5]: @paulbourelly999 Addressed

File: streets_utils/streets_desired_phase_plan/cmake/streets_desired_phase_plan_libConfig.cmake.in (hunk 3 above)
Comment (paulbourelly999) [reply streets_utils/streets_desired_phase_plan/cmake/streets_desired_phase_plan_libConfig.cmake.in:6]: but they are not part of the library, they are part of the test executable right.?

Comment (dan-du-car) [reply streets_utils/streets_desired_phase_plan/cmake/streets_desired_phase_plan_libConfig.cmake.in]: updated

**Review by Saeid-Solei — APPROVED** [review]

**Review by paulbourelly999 — APPROVED** [review]
Looks good.

**Pull request merged**